### PR TITLE
chore: add constraint to resolution of requirements-dev.txt

### PR DESCRIPTION
**Pull request by LeonLuttenberger**
*Description of changes:*
Our `requirements-dev.txt` resolution doesn't take into account our production dependencies.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/requirements-docs.txt [hunk 1]
```diff
@@ -1,8 +1,8 @@
 #
-# This file is autogenerated by pip-compile with Python 3.9
+# This file is autogenerated by pip-compile with Python 3.8
 # by the following command:
 #
-#    pip-compile requirements-docs.in
+#    pip-compile --constraint=requirements.txt --output-file=docs/requirements-docs.txt docs/requirements-docs.in
```
Comment (dgraeber) [inline docs/requirements-docs.txt:5]: why is this needed?  This reqs file is only for generating the readthedocs site...(ref seed-farmer/.readthedocs.yaml)

**Review by dgraeber — COMMENTED** [review]
I am not sure what the benefit of the PR is (just looking for a reason to do this....not opposed to it otherwise)

**Comment by LeonLuttenberger** [comment]
> I am not sure what the benefit of the PR is (just looking for a reason to do this....not opposed to it otherwise)

Mainly just to keep the requirements consistent. A minor concern is that in our workflow, we first install from `requirement.txt` and then from `requirements-dev.txt`. In the latter, we have outdated versions of certain packages that overlap. So our workflows end up with outdated versions of certain packages, instead of testing on the one we specify in `requirement.txt`. Dependabot doesn't seem to keep this in line either, although that could change now that the `pip-compile` command includes the constraints argument.

This, however, is quite a minor concern. The main thing is just keep our dependencies clean and consistent.

**Comment by dgraeber** [comment]
> > I am not sure what the benefit of the PR is (just looking for a reason to do this....not opposed to it otherwise)
> 
> Mainly just to keep the requirements consistent. A minor concern is that in our workflow, we first install from `requirement.txt` and then from `requirements-dev.txt`. In the latter, we have outdated versions of certain packages that overlap. So our workflows end up with outdated versions of certain packages, instead of testing on the one we specify in `requirement.txt`. Dependabot doesn't seem to keep this in line either, although that could change now that the `pip-compile` command includes the constraints argument.
> 
> This, however, is quite a minor concern. The main thing is just keep our dependencies clean and consistent.

Ok...just remove the change to the docs/requirements-docs.txt -- that is really only for readthedocs.  I like this change for the requirments-dev.txt though....good catch!   I will look to remove the sphinx references from the requirements-dev.txt also...that is no longer needed (I believe)

**Review by dgraeber — APPROVED** [review]
